### DISCUSSION TITLE
sending confirmation email in error

### DISCUSSION THREAD

**Discussion by ewo68** [pglombardo/PasswordPusher#2034] [General]
Hi, although all the values are entered for SMTP, when I create a user I do not receive the email. If I go to "Didn't receive confirmation instructions?" after typing the email I got an error 500.

Would you know how to troubleshoot this?
![image](https://github.com/pglombardo/PasswordPusher/assets/107499018/b8f0caf7-95bf-48f0-8687-b5598b7ce682)

Thanks

**Comment by pglombardo** [comment]
Hi @ewo68 - in v1.39.8, I added a tool to test email configurations.  Could you update to that version and try [this out](https://github.com/pglombardo/PasswordPusher/wiki/How-to-Validate-and-Test-Email-Settings)?

It should reveal where the issue is...

**Comment by ewo68** [comment]
Hello, thank you for your feedback.
I am in version 1.39.7

In the documentation it tells me to go to ./bin.pwpush console


Otherwise it doesn't work. By connecting to ./bin/raise console when I do the test I get this error.


And yet I made sure to PWP__MAIL__RAISE_DELIVERY_ERRORS=true

I am docker under Synology


Here is the SMTP information.
]

THANKS
​​​​​
​Cordialement,
​

**Reply by pglombardo** [reply]
Not all the images are coming through so I didn't get all of the info - but do upgrade to 1.39.8 - it has new code that will show the exact error for the email issue.

I'll update the `stable` tag to point to `1.39.8` soon.

**Comment by ewo68** [comment]
thank you for your comeback. I installed version 1.39.8, and enabled PWP__MAIL__RAISE_DELIVERY_ERRORS=true
But when I run for example the command
I made a short video so you can see the error. Will try it after the SSH command

Video :

**Comment by ewo68** [comment]
oh yes the test mailer passed :

**Reply by pglombardo** [reply]
Ok that means that the issue may likely be in the `mailer_sender`.  In that `pwpush console`, what is the output of `Settings.mail.mailer_sender`?

Additionally, could you find the error in `log/production.log` and post the entire error message here?

**Comment by ewo68** [comment]
sorry I did not understand "output of Settings.mail.mailer_sender?"

I think this is it

***@***.***:~# docker exec -it <ID> bash
PasswordPusher:/opt/PasswordPusher$ ./bin/rails console
Password Pusher Version: 1.39.8
Loading production environment (Rails 7.1.3.2)
irb(main):001> Rails.application.config.action_mailer.smtp_settings
=>
{:address=>"smtp.fr",
 :port=>25,
 ***@***.***",
 :password=>"mdpusersmtp",
 :authentication=>"plain",
 :enable_starttls_auto=>true,
 :open_timeout=>10,
 :read_timeout=>10}
irb(main):002>


Test error 500 result production.log

PasswordPusher:/opt/PasswordPusher/log$ head production.log
# Logfile created on 2024-04-09 12:29:49 +0000 by logger.rb/v1.5.3
E, [2024-04-09T12:32:45.090612 #15] ERROR -- : [b6cd2cbb-6634-498a-a15f-ece78e2daaa6]
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6] ActionView::Template::Error (Missing host to link to! Please provide the :host parameter,                                                                                                 set default_url_options[:host], or set :only_path to true):
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     21:
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     22: <p>
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     23: <%= _('You can confirm your account email by') %>
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     24: <%= link_to _('by clicking this link'), ***@***.***, confirmation_                                                                                                token: @token) %>
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     25: <%= _('or by copying and pasting the following URL into your browser:') %>
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     26: </p>
[b6cd2cbb-6634-498a-a15f-ece78e2daaa6]     27:




result test testmail
irb(main):001> ***@***.***").deliver_now
=> #<Mail::Message:332460, Multipart: false, Headers: <Date: Tue, 09 Apr 2024 12:35:04 +0000>, <From: ***@***.***>, <To: ***@***.***>, <Message-ID: ***@***.***>>, <Subject: Test Email from Password Pusher>, <Mime-Version: 1.0>, <Content-Type: text/plain>, <Content-Transfer-Encoding: quoted-printable>>
​​​​​
​

**Reply by pglombardo** [reply]
That output helps.  Try setting these [two values](https://github.com/pglombardo/PasswordPusher/blob/master/config/settings.yml#L107-L125).  That should fix the error you posted.

**Reply by ewo68** [reply]
oh great, it works, it was the PWP__HOST_DOMAIN value that was missing.
Well done, and thank you
Best wishes.

**Reply by pglombardo** [reply]
Excellent 😎
